app/main/views/index.py

```diff
@@ -19,12 +19,17 @@ def index():
     if subscription_form.validate_on_submit():
         return redirect(url_for('main.subscription', email=subscription_form.email.data))
 
+    past_events = api_client.get_events_past_year()
+    all_events = future_events + past_events[::-1]
```

Comment:
Rather than making a call to past events regardless of whether they are needed it would be better to only make the call if there are less than 3 future events.

Also I would limit the number of past events passed to be just what is required.

It should also be possible for each past event to add in a property to mark it as being a past event - 
Something like - 

```
for event in past_events:
  event['past'] = True
```

Then in the template you can check if the `past` flag is set.

Reply:
When it comes to testing, whats the appropriate way to do so? At the moment a mock `sample_future_events` gets passed into the test and is iterated through. So should i create an `sample_all_events` that has an extra field set to `past` on some of the events and when i iterate through `sample_all_events` I can just find the ones with past and check that it can be found on the page?

Reply:
That seems like a way to go, you want to make sure that for past events there is the additional style to indicate its status and also that the carousel only shows future events unless there are non in which case I guess it could show the last 3 past events?